src/java.compiler/share/classes/javax/tools/StandardJavaFileManager.java

```diff
@@ -267,7 +267,8 @@
      * @throws IllegalArgumentException if the array of files includes
      * a directory
      * @throws NullPointerException if the given array contains null
-     * elements
+     * elements and the given element is used by
+     * {@linkplain #getJavaFileObjectsFromPaths(Collection)}.
```

Comment:
The new condition should be in an `@implSpec`.

The package declaration contains a general specification for nulls and NPE, so it's not clear that we need the `@throws NPE` at all, but it's not wrong to leave it as was.